src/bindings/python/src/pyopenvino/graph/op_extension.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/op_extension.hpp"
+
+#include <pybind11/pybind11.h>
+
+#include <pyopenvino/graph/op_extension.hpp>
+
+#include "pyopenvino/core/common.hpp"
+#include "pyopenvino/core/extension.hpp"
+#include "pyopenvino/graph/discrete_type_info.hpp"
+#include "pyopenvino/graph/node_output.hpp"
+#include "pyopenvino/graph/op.hpp"
+
+namespace py = pybind11;
+
+const ov::DiscreteTypeInfo& PyOpExtension::get_type_info() const {
+    return *m_type_info;
+}
+
+ov::OutputVector PyOpExtension::create(const ov::OutputVector& inputs, ov::AttributeVisitor& visitor) const {
+    py::gil_scoped_acquire acquire;
+
+    const auto node = py_handle_dtype();
+
+    node.attr("set_arguments")(py::cast(inputs));
```

Comment:
Since we already have a wrapper here, maybe we could somehow migrate all the manual usage of `constructor_validate_and_infer_types` from public API and hide it here?

What I mean is the ideal scenario in which the user only has to define `constructor_validate_and_infer_types` but never has to call it themselves. Also could be solved by Sergey's idea.